tools/docker/static-analysis/cppcheck.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+scriptdir="$(cd "${0%/*}"; pwd)"
+rootdir="${scriptdir%/*/*/*}"
+
+. "${rootdir}/tools/docker/functions.sh"
+
+OUTPUT_FILE="$rootdir/cppcheck_results.txt"
```

Comment:
why not add the output in an 'out' directory in this folder? (nitpick)

Reply:
We could, although we don't have such a folder for anything else at the moment.
In our case the scrupt is mostly meant to be used in a docker container though, because the results highly depend on the `cppcheck` version.